algorithms/schema-free/greedy_graph_coloring.gsql

```diff
@@ -0,0 +1,78 @@
+CREATE QUERY greedy_graph_coloring(SET<STRING> v_type,SET<STRING> e_type,BOOL print_color_count = TRUE,BOOL display = TRUE, STRING file_path = ""){
+  
+  /* Greedy Graph Coloring algorithm
+  This is a distributed algorithm for coloring graph on large-scale undirect graphs with     one edge type.In every iteration, the vertices send its color to its out-neighbors, and     the neighbors get assigned a different color. 
+  Conditions:
+    -  No two adjacent vertices in the graph should have same color
+    -  Until above condition is achieved , keep on finding conflicts and resolve them.
+    -  Stop the conflict detection when there are no resolution left.
+  */
+  
+  TYPEDEF tuple<INT color> testColor;
+  AndAccum<BOOL> @@isconflict=true;
+  HeapAccum<testColor>(999999,color ASC) @color;
+  SumAccum<INT> @colorvertex;
+  MaxAccum<INT> @@counter;
+  SetAccum<EDGE> @@edgeSet;
+  SetAccum<Vertex> @@vertexSet;
+  FILE f(file_path);
+  
+  #initialization - coloring All vertices
+  coloring = {v_type};
+  coloring = SELECT s 
+             FROM coloring:s 
+             POST-ACCUM s.@colorvertex+=1,@@counter+=s.@colorvertex;
+  
+  #conflict detection and resolution
+  WHILE @@isconflict DO
+    start= {v_type};
+  
+    #clear the color heap for all vertices
+    start = SELECT s 
+            FROM start:s
+            POST-ACCUM s.@color.clear();
+  
+    #create a color heap for all vertices
+    start = SELECT t 
+            FROM start:s-(e_type:e)-:t
+            ACCUM t.@color+=testColor(s.@colorvertex);
```

Comment:
Technically, the FROM pattern should be start:s-(e_type:e)-v_type:t, with an explicit vertex type for the target vertices.  If the edge type is a wildcard, or has a wildcard for its target vertex type, then omitting v_type from the target will yield the wrong answer.